pkg/cluster/deploybaseresources_additional.go

```diff
@@ -532,11 +529,36 @@ func (m *manager) networkPublicLoadBalancer(azureRegion string) *arm.Resource {
 		})
 	}
 
-	return &arm.Resource{
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs != nil {
+		for i := len(*lb.FrontendIPConfigurations); i < m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs.Count; i++ {
+			resourceGroupID := m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID
+			frontendIPConfigName := stringutils.LastTokenByte(outboundIPs[i].ID, '/')
+			frontendConfigID := fmt.Sprintf("%s/providers/Microsoft.Network/loadBalancers/%s/frontendIPConfigurations/%s", resourceGroupID, *lb.Name, frontendIPConfigName)
+			*lb.FrontendIPConfigurations = append(*lb.FrontendIPConfigurations, newFrontendIPConfig(frontendIPConfigName, frontendConfigID, outboundIPs[i].ID))
+		}
+
+		for i := 0; i < m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs.Count; i++ {
+			resourceGroupID := m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID
+			if i == 0 && m.doc.OpenShiftCluster.Properties.APIServerProfile.Visibility == api.VisibilityPublic {
+				frontendIPConfigName := "public-lb-ip-v4"
+				frontendConfigID := fmt.Sprintf("%s/providers/Microsoft.Network/loadBalancers/%s/frontendIPConfigurations/%s", resourceGroupID, *lb.Name, frontendIPConfigName)
+				*(*lb.OutboundRules)[0].FrontendIPConfigurations = append(*(*lb.OutboundRules)[0].FrontendIPConfigurations, newOutboundRuleFrontendIPConfig(frontendConfigID))
+				continue
+			}
+			frontendIPConfigName := stringutils.LastTokenByte(outboundIPs[i].ID, '/')
+			frontendConfigID := fmt.Sprintf("%s/providers/Microsoft.Network/loadBalancers/%s/frontendIPConfigurations/%s", resourceGroupID, *lb.Name, frontendIPConfigName)
+			*(*lb.OutboundRules)[0].FrontendIPConfigurations = append(*(*lb.OutboundRules)[0].FrontendIPConfigurations, newOutboundRuleFrontendIPConfig(frontendConfigID))
+		}
+	}
+
+	armResource := &arm.Resource{
 		Resource:   lb,
 		APIVersion: azureclient.APIVersion("Microsoft.Network"),
-		DependsOn: []string{
-			"Microsoft.Network/publicIPAddresses/" + m.doc.OpenShiftCluster.Properties.InfraID + "-pip-v4",
-		},
+		DependsOn:  []string{},
+	}
+	for _, ip := range outboundIPs {
+		ipName := stringutils.LastTokenByte(ip.ID, '/')
+		armResource.DependsOn = append(armResource.DependsOn, "Microsoft.Network/publicIPAddresses/"+ipName)
```

Comment:
I think there's a bug here. If API visibility is public and `ManagedOutboundIPs` is nil, `outboundIPs` will also be `nil`, so the LB won't be set up to depend on the `-pip-v4` PIP, which it still needs. Does this need to be reworked to ensure that the LB depends on the `-pip-v4` PIP whenever API visibility is public?

Reply:
Good catch, when adding outboundIPs/outboundIPPrefixes to this feature there will be a bug if that IP is not added.  I'll update this to prevent that from happening.